Maze.cpp

```diff
@@ -114,7 +117,7 @@ Path Maze::findPath(CellCoordinate start, CellCoordinate end, Direction facing )
 			}
 
 			// This path is the best so far.
-			adjacentNode->previous = *currentNodeItr;
+			(*currentNodeItr)->next = adjacentNode;
```

Comment:
This seems to work just fine. The only negative is that we lose the sometimes useful property that any node touched by pathfinding will also store the shortest path to start.
Seen here: 
![image](https://cloud.githubusercontent.com/assets/4276762/23532021/28137752-ff5e-11e6-9ef0-15b62e604c34.png)

I think the best of both worlds would be to revert to using the previous pointers, but call the pathfinding in reverse. That way the path is constructed in the order it will be followed and all neighboring nodes will contain the information need to construct a new shortest path without going through the entire pathfind again.